monai/bundle/scripts.py

```diff
@@ -52,12 +55,61 @@ def _update_args(args: Optional[Union[str, Dict]] = None, ignore_none: bool = Tr
 
 
 def _log_input_summary(tag: str, args: Dict):
+    """
+    Log the arguments of bundle scripts.
+
+    Args:
+        tag: tag to identify the script in the log.
+        args: arguments of the script to log.
+
+    """
     logger.info(f"\n--- input summary of monai.bundle.scripts.{tag} ---")
     for name, val in args.items():
         logger.info(f"> {name}: {pprint.pformat(val)}")
     logger.info("---\n\n")
 
 
+def _get_var_names(expr: str):
+    """
+    Parse the expression and discover what variables are present in it based on ast module.
+
+    Args:
+        expr: source expression to parse.
+
+    """
+    tree = ast.parse(expr)
+    return [m.id for m in ast.walk(tree) if isinstance(m, ast.Name)]
+
+
+def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int = 1, any: int = 1) -> Tuple:
+    """
+    Get spatial shape for fake data according to the specified shape pattern.
+    It supports `int` number and `string` with formats like: "32", "32 * n", "32 ** p", "32 ** p *n".
+
+    Args:
+        shape: specified pattern for the spatial shape.
+        p: power factor to generate fake data shape if dim of expected shape is "x**p", default to 1.
+        p: multiply factor to generate fake data shape if dim of expected shape is "x*n", default to 1.
+        any: specified size to generate fake data shape if dim of expected shape is "*", default to 1.
+
+    """
+    ret = []
+    for i in shape:
+        if isinstance(i, int):
+            ret.append(i)
+        elif isinstance(i, str):
+            if i == "*":
+                ret.append(any)
+            else:
+                for c in _get_var_names(i):
+                    if c not in ["p", "n"]:
+                        raise ValueError(f"only support variables 'm' and 'p' so far, but got: {c}.")
```

Comment:
The error msg has a typo m-> n?

Reply:
Thanks for your finding, updated in the latest commit.